client/my-sites/site-settings/traffic/controller.js

```diff
@@ -11,40 +11,29 @@ import analytics from 'lib/analytics';
 import { renderWithReduxStore } from 'lib/react-helpers';
 import route from 'lib/route';
 import TrafficMain from 'my-sites/site-settings/traffic/main';
-import sitesFactory from 'lib/sites-list';
-import utils from 'lib/site/utils';
-
-const sites = sitesFactory();
+import { getSelectedSite, getSelectedSiteId } from 'state/ui/selectors';
+import { canCurrentUser } from 'state/selectors';
 
 export default {
 	traffic( context ) {
 		const analyticsPageTitle = 'Site Settings > Traffic';
 		const basePath = route.sectionify( context.path );
-		const fiveMinutes = 5 * 60 * 1000;
-		let site = sites.getSelectedSite();
+		const state = context.store.getState();
+		const site = getSelectedSite( state );
+		const siteId = getSelectedSiteId( state );
+		const canManageOptions = canCurrentUser( state, siteId, 'manage_options' );
 
 		// if site loaded, but user cannot manage site, redirect
-		if ( site && ! utils.userCan( 'manage_options', site ) ) {
+		if ( site && ! canManageOptions ) {
 			page.redirect( '/stats' );
 			return;
 		}
 
-		if ( ! site.latestSettings || new Date().getTime() - site.latestSettings > ( fiveMinutes ) ) {
-			if ( sites.initialized ) {
-				site.fetchSettings();
-			} else {
-				sites.once( 'change', function() {
-					site = sites.getSelectedSite();
-					site.fetchSettings();
```

Comment:
@gwwar the `<QuerySiteSettings />` was introduced recently, while the `site.fetchSettings()` and `site.saveSettings()` have been there since the creation of SEO settings.

The `<QuerySiteSettings />` was introduced in #11788 and one of the next goals was to completely get sites-list out of the Traffic tab. So this PR is really welcome 👍